modules/project_cleanup/function_source/main.go

```diff
@@ -611,6 +624,33 @@ func invoke(ctx context.Context) {
 		return err
 	}
 
+	removeProjectClusters := func(projectId string) int {
+		logger.Printf("Try to remove clusters for [%s]", projectId)
+		reqLCR := &containerpb.ListClustersRequest{Parent: fmt.Sprintf("projects/%s/locations/*", projectId)}
+		listResponse, err := containerService.ListClusters(ctx, reqLCR)
+		if err != nil {
+			logger.Printf("Failed to list clusters for [%s], error [%s]", projectId, err.Error())
+			return 0
+		}
+
+		logger.Printf("Got [%d] clusters for project [%s]", len(listResponse.Clusters), projectId)
+		if len(listResponse.Clusters) == 0 {
+			return 0
+		}
+
+		for _, cluster := range listResponse.Clusters {
+			logger.Printf("Try to remove cluster: %s", cluster.Name)
+
+			reqDCR := &containerpb.DeleteClusterRequest{Name: fmt.Sprintf("projects/%s/locations/%s/clusters/%s", projectId, cluster.Location, cluster.Name)}
+			_, err := containerService.DeleteCluster(ctx, reqDCR)
+			if err != nil {
+				logger.Printf("Failed to delete cluster [%s] for [%s], error [%s]", cluster.Name, projectId, err.Error())
+			}
+		}
+
+		return len(listResponse.Clusters)
```

Comment:
Could there be a possibility that len > 0 but all clusters got an err response for `DeleteCluster` so it won't move forward? My thinking is we could return `len(listResponse.Clusters)-len(deleteFailures)`

Reply:
Yeah - I decided to implement a new `pendingDeletion` int using the switch case to provide finer grain granularity.